sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -837,8 +837,36 @@ case class Cast(child: Expression, dataType: DataType) extends UnaryExpression w
     val j = ctx.freshName("j")
     val values = ctx.freshName("values")
 
+    val isPrimitiveFrom = ctx.isPrimitiveType(fromType)
+    val isPrimitiveTo = ctx.isPrimitiveType(toType)
+
     (c, evPrim, evNull) =>
-      s"""
+      if (isPrimitiveFrom && isPrimitiveTo) {
+        // ensure no null in input and output arrays
```

Comment:
I assume that here is a part to generate code for Java primitive arrays regarding ```from``` and ```to```. Since Java primitive array (e.g. int[]) cannot have ```null``` value unlike SQL, I said "ensure not null in input and output arrays."
What do you think?